src/NUnitEngine/nunit.engine/Services/ExtensionService.cs

```diff
@@ -113,18 +113,26 @@ public ExtensionNode GetExtensionNode(string path)
             return ep != null && ep.Extensions.Count > 0 ? ep.Extensions[0] : null;
         }
 
-        public IEnumerable<ExtensionNode> GetExtensionNodes<T>()
+        public IEnumerable<ExtensionNode> GetExtensionNodes<T>(bool includeDisabled = false)
         {
             var ep = GetExtensionPoint(typeof(T));
             if (ep != null)
                 foreach (var node in ep.Extensions)
-                    yield return node;
+                    if (includeDisabled || node.Enabled)
+                        yield return node;
         }
 
         public IEnumerable<T> GetExtensions<T>()
         {
             foreach (var node in GetExtensionNodes<T>())
-                yield return (T)node.ExtensionObject;
+              	 yield return (T)node.ExtensionObject;
+        }
+
+        public void EnableExtension(string typeName)
+        {
+            foreach (var node in Extensions)
+                if (node.TypeName == typeName)
```

Comment:
Would be nice to have Linq in the engine. Stuff like this would be cleaner with a `.Where()`. :smile: